Examples/VoiceMemos/VoiceMemosTests/VoiceMemosTests.swift

```diff
@@ -124,6 +124,7 @@ class VoiceMemosTests: XCTestCase {
     )
 
     store.send(.recordButtonTapped)
```

Comment:
`.recordButtonTapped` ends up returning https://github.com/pointfreeco/swift-composable-architecture/blob/313dd217dcd1d0478118ec5d15225fd473c1564a/Examples/VoiceMemos/VoiceMemos/VoiceMemos.swift#L70-L76
using `RunLoop.immediate` results in a failed test because `.currentRecordingTimerUpdated` now also gets received once before `.audioRecorder(.failure(.couldntActivateAudioSession))` is received. I'm unsure if this is an undesired result of the changed.

Adding following test  to `TimerTests.swift` demonstrates the difference in behaviour.

```swift
 func testTimerImmediate() {
    let scheduler = DispatchQueue.immediate

    var count = 0

    Effect.timer(id: 1, every: .seconds(1), on: scheduler)
      .sink { _ in count += 1 }
      .store(in: &self.cancellables)

    XCTAssertNoDifference(count, 1) // This PR

    // XCTAssertNoDifference(count, 0) // Current implementation
  }
```

Reply:
For the current implementation, the timer is triggered as soon as the `PassThroughSubject` subscribes to it, essentially dropping the value before the effect is subscribed to downstream. 

https://github.com/pointfreeco/swift-composable-architecture/blob/313dd217dcd1d0478118ec5d15225fd473c1564a/Sources/ComposableArchitecture/Effects/Cancellation.swift#L36-L37

So the timer auto connects and outputs value at line 37 which is lost before returning the effect.

In this PR the timer auto connects at `.sink { _ in count += 1 }`

Reply:
Interesting find! I think overall this PR is an improvement in the expectation that at least 1 thing emits from a timer running on an immediate scheduler, even if it is a breaking change.